lib/iris/_cube_coord_common.py

```diff
@@ -177,6 +177,22 @@ def _check(item):
 
         return result
 
+    @property
+    def names(self):
+        """
+        A tuple containing all of the metadata names. This includes the
+        standard name, long name, NetCDF variable name, and attributes
+        STASH name.
+
+        """
+        standard_name = self.standard_name
+        long_name = self.long_name
+        var_name = self.var_name
+        stash_name = self.attributes.get('STASH')
+        if stash_name is not None:
+            stash_name = str(stash_name)
+        return (standard_name, long_name, var_name, stash_name)
```

Comment:
I'm not sure about the addition of this `names` method on CFVariableMixin. 

`stash_name` makes sense for a cube but not a coordinate (or cell measure or ancillary variable  etc )

Reply:
The introduction of the `names` property is in total alignment with the existing `name` method and its behaviour. 

The `name` method also doesn't make sense for anything other than a `cube`, and yet we have it applied to coordinates and else where. So your argument also applies to that, and yet it still stands. This PR is highlighting existing behaviour already available in `iris` - to the surprise of @pp-mo.

So either we honour this relationship, or we remove the use of `STASH` from both the `names` property and the `name` method. If we did, then that would be a breaking change for the API, rather than an additive change.